StonehearthEditor/Recipes/RecipesView.cs

```diff
@@ -0,0 +1,701 @@
+            if (recipesIndex == null)
+                return;
+
+            Dictionary<string, FileData> recipeFileData = recipesIndex.LinkedFileData;
+            foreach (KeyValuePair<string, FileData> recipe in recipeFileData)
+            {
+                JsonFileData jsonFileData = recipe.Value as JsonFileData;
+                JObject recipeJson = jsonFileData.Json;
+                JArray ingredientArray = recipeJson["ingredients"] as JArray;
+                JArray productArray = recipeJson["produces"] as JArray;
+
+                foreach (JToken product in productArray)
+                {
+                    RecipeRow row = mDataTable.NewRecipeRow();
+
+                    row.RecipeList = recipesIndex;
+                    row.Recipe = jsonFileData;
+
+                    JToken lvlReq = recipeJson["level_requirement"];
+                    row.SetLevelRequired(lvlReq == null ? 0 : lvlReq.ToObject<int>());
+                    row.SetEffort(recipeJson["work_units"].ToObject<int>());
+                    row.SetCrafter(jobAlias);
+
+                    // TODO: make a row for each fine item?
+                    JToken item = product["item"];
+                    if (item != null)
+                    {
+                        string alias = item.ToString();
+                        row.SetAlias(alias);
+
+                        // Check aliases linked by recipe file
+                        JsonFileData itemFileData = FindJsonFileMatchingAlias(jsonFileData, alias);
+                        row.Item = itemFileData;
+                        row.SetNetWorth(itemFileData.NetWorth);
+                        row.SetDisplayName(GetTranslatedName(GetDisplayName(itemFileData)));
+                        row.SetIcon(GetIcon(itemFileData));
+                    }
+
+                    foreach (JToken ingredient in ingredientArray)
+                    {
+                        JToken uri = ingredient["uri"];
+                        JToken material = ingredient["material"];
+                        JToken count = ingredient["count"];
+
+                        Ingredient ingredientData = row.AddNewIngredient();
+
+                        string alias = uri != null ? uri.ToString() : material.ToString();
+
+                        ingredientData.Amount = count.ToObject<int>();
+
+                        if (material != null)
+                        {
+                            ingredientData.Name = alias;
+                            ingredientData.Icon = GetIcon(material.ToString());
+                        }
+                        else if (uri != null)
+                        {
+                            JsonFileData ingrJsonFileData = FindJsonFileMatchingAlias(jsonFileData, alias);
+                            ingredientData.Name = ingrJsonFileData.GetModuleFile().FullAlias;
+                            ingredientData.Icon = GetIcon(ingrJsonFileData);
+                        }
```

Comment:
The else case here should show an error, shouldn't it?

Reply:
yes